synapse/util/__init__.py

```diff
@@ -25,9 +25,18 @@
 
 logger = logging.getLogger(__name__)
 
+
+def _reject_invalid_json(val):
+    """Do not allow Infinity, -Infinity, or NaN values in JSON."""
+    raise json.JSONDecodeError("Invalid JSON value: '%s'" % val)
+
+
 # Create a custom encoder to reduce the whitespace produced by JSON encoding.
 json_encoder = json.JSONEncoder(separators=(",", ":"))
```

Comment:
Can we also do the same here and set `allow_nan=False` to stop us from sending invalid json too?

Reply:
Good call! 👍

Reply:
Done in 248d8284fa850289689f9ae87d2c807b58d7a812.

There's a few places we don't use this encoder, I was going to move those over in a follow-up (but I can do it here if you'd like).

Reply:
(Hmm...we should probably make this change to canonicaljson too. I'll file an issue there.)

Reply:
See matrix-org/python-canonicaljson#32.